Cms/ProfileBioEdit.ascx.cs

```diff
@@ -0,0 +1,2394 @@
+                        Number = phoneNumber.NumberFormatted,
+                        RequiredErrorMessage = $"{phoneNumber.NumberTypeValue.Value} phone is required",
+                        Required = requiredPhoneTypes.Contains( phoneNumber.NumberTypeValue.Guid ) && !isChild,
+                        ValidationGroup = BlockValidationGroup
+                    };
+                    var cbSms = new RockCheckBox
+                    {
+                        ID = $"cbSms{phoneNumber.NumberTypeValueId}",
+                        Text = GetAttributeValue( AttributeKey.SMSEnableLabel ),
+                        Checked = phoneNumber.IsMessagingEnabled,
+                        ContainerCssClass = "mb-0",
+                        CssClass = "js-sms-number",
+                        Visible = GetAttributeValue( AttributeKey.ShowSMSEnable ).AsBoolean() && phoneNumber.NumberTypeValueId == Rock.Web.Cache.DefinedValueCache.GetId( new Guid( Rock.SystemGuid.DefinedValue.PERSON_PHONE_TYPE_MOBILE ) )
+                    };
+                    lblPhone.AssociatedControlID = pnbPhone.ID;
+
+                    if ( pnbPhone.Required )
+                    {
+                        pnlPhoneNumContainer.AddCssClass( "required" );
+                    }
+
+                    pnlPhoneNumControls.Controls.Add( hfPhoneType );
+                    pnlPhoneNumControls.Controls.Add( pnbPhone );
+                    pnlPhoneNumControls.Controls.Add( cbSms );
+
+                    contactFieldsControls.Add( pnlPhoneNumContainer );
+                }
+            }
+
+            rblEmailPreference.Items.Clear();
+            rblEmailPreference.Items.Add( new ListItem( "All Emails", "EmailAllowed" ) );
+            rblEmailPreference.Items.Add( new ListItem( "Only Personalized", "NoMassEmails" ) );
+            rblEmailPreference.Items.Add( new ListItem( "Do Not Email", "DoNotEmail" ) );
+            rblEmailPreference.RepeatDirection = RepeatDirection.Horizontal;
+            rblEmailPreference.SelectedValue = person.EmailPreference.ConvertToString( false );
+
+            rblCommunicationPreference.Items.Clear();
+            rblCommunicationPreference.Items.Add( new ListItem( "Email", "1" ) );
+            rblCommunicationPreference.Items.Add( new ListItem( "SMS", "2" ) );
+            rblCommunicationPreference.RepeatDirection = RepeatDirection.Horizontal;
+            rblCommunicationPreference.SetValue( person.CommunicationPreference == CommunicationType.SMS ? "2" : "1" );
+
+            if ( isChild )
+            {
+                var displayEmailPreference = GetAttributeValue( AttributeKey.EmailPreference );
+                rblEmailPreference.Visible = displayEmailPreference == "Show on All";
+
+                var displayCommunicationPreference = GetAttributeValue( AttributeKey.CommunicationPreference );
+                rblCommunicationPreference.Visible = displayCommunicationPreference == "Show on All";
```

Comment:
Test if changing between role updates visibility of communication preferences properly.

Reply:
It was not quite working properly when selecting a different role, but for other reasons, it was actually the column was still showing when it shouldn't.